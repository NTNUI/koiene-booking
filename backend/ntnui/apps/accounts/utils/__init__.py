from accounts.utils.mail_utils import send_gsuite_confirmation_mail
