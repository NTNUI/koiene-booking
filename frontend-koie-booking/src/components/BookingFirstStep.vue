<template>
  <ErrorCard v-if="apiError" />
  <LoadingSpinner v-else-if="isLoading" />
  <v-layout v-else :class="$style.container" :dark="true">
    <h1 :class="$style.heading">{{ $t('booking.reservation') }} {{ koieTitle }}</h1>
    <v-layout :class="$style.separator">
      <h3>{{ $t('booking.dates') }}</h3>
      <v-layout class="px-1">
        <Calendar :vbeds="false" :row="true" />
      </v-layout>
    </v-layout>
    <v-layout :class="$style.separator">
      <h3>{{ $t('booking.bed_reservation') }}</h3>
      <v-layout :class="$style.availableBeds">
        <v-icon :class="$style.icon" v-text="'mdi-bed'"></v-icon>
        <p :class="$style.availableBedsText">
          {{ bedsAvailable }} / {{ bedsTotal }} {{ $t('booking.available_beds') }}
        </p>
      </v-layout>
      <v-layout :class="$style.chooseBeds">
        <p style="margin-bottom: 0" :class="$style.chooseText">{{ $t('booking.choose_numbers') }}</p>
        <v-text-field
          v-model="bedsChosen"
          :value="bedsChosen"
          :class="$style.bedPicker"
          min="1"
          :max="bedsAvailable"
          :dense="true"
          hide-details
          single-line
          outlined
          type="number"
          @change="(e) => setBeds(e)"
        />
      </v-layout>
      <v-alert
        v-if="error"
        v-model="error"
        transition="expand-transition"
        :dismissible="true"
        :class="$style.errorMsg"
        type="info"
        @click="error = false"
      >
        {{ $t('calendar.errormsg') }} {{ bedsAvailable }}
      </v-alert>
    </v-layout>
  </v-layout>
</template>

<script lang="ts">
import Calendar from '../components/Calendar.vue';
import ErrorCard from '../components/ErrorCard.vue';
import LoadingSpinner from '../components/LoadingSpinner.vue';
import Vue from 'vue';
import { BookingFirstStepData } from '../types/booking';
import BookingSecondStepVue from './BookingSecondStep.vue';

export default Vue.extend({
  name: 'BookingFirstStepVue',
  components: {
    Calendar,
    ErrorCard,
    LoadingSpinner
  },
  data(): BookingFirstStepData {
    return {
      bedsTotal: 8,
      bedsChosen: 1,
      error: false
    };
  },
  computed: {
    koieTitle(): string {
      return this.$store.state.koie.koieData.name || this.$route.params.id;
    },
    apiError(): boolean {
      return this.$store.state.koie.error;
    },
    isLoading(): boolean {
      return this.$store.state.koie.isLoading;
    },
    bedsAvailable(): number {
      return this.$store.state.booking.availableBeds;
    }
  },
  created() {
    this.getTotalBeds();
    if (this.$store.state.booking.beds <= this.bedsAvailable) {
      this.bedsChosen = this.$store.state.booking.beds;
    }
  },
  methods: {
    setBeds(number: number) {
      if (this.$store.state.booking.edited) {
        this.$store.dispatch('booking/SET_EDITED', false);
      }
      if (number <= this.bedsAvailable) {
        this.$store.dispatch('booking/SET_BEDS', number);
      } else {
        this.bedsChosen = 1;
        this.$store.dispatch('booking/SET_BEDS', 1);
        this.error = true;
      }
    },
    goToNextStep() {
      this.$store.dispatch('SET_STEP', 2);
    },
    async getTotalBeds() {
      if (this.$store.state.koie.koieData.name.toLowerCase() !== this.koieTitle.toLowerCase()) {
        await this.$store.dispatch('koie/FETCH_DATA', this.koieTitle.toLowerCase());
      }
      this.bedsTotal = await this.$store.state.koie.koieData.number_of_beds;
    }
  }
});
</script>

<style lang="scss" module>
.container {
  display: flex;
  flex-direction: column;
}
.heading {
  margin-top: 20px;
  align-self: center;
}
.separator {
  border-left: solid 4px;
  border-color: #2f3a50;
  display: flex;
  flex-direction: column;
  margin-top: 24px;
}

.separator > h3 {
  padding: 16px;
}
.icon {
  margin-right: 8px;
  margin-bottom: 6px;
}
.availableBeds {
  display: flex;
  flex-direction: row;
  padding-left: 16px;
  padding-right: 16px;
}
.availableBedsText {
  margin-top: 10px;
}
.chooseBeds {
  display: flex;
  align-items: center;
  padding-left: 16px;
  padding-right: 16px;
}
.chooseText {
  margin-right: 20px;
}
.bedPicker {
  max-width: 60px;
  margin-left: 20px;
}
.errorMsg {
  margin-top: 10px;
}
</style>
