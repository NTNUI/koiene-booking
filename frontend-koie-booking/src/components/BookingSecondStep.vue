<template>
  <v-layout :class="$style.container">
    <v-layout :class="$style.separator">
      <h3 class="py-4" :class="$style.guestRegistration">{{ $t('booking.guest_registration') }}</h3>
      <v-form v-model="validForm" :class="$style.form">
        <v-layout v-for="i in Number(bedsChosen)" :key="i">
          <div :class="$style.fieldWrapper">
            <v-text-field
              v-if="i === 1"
              v-model="guests[i - 1].name"
              :label="$t('booking.full_name')"
              placeholder="Ola Nordmann"
              :rules="nameRules"
              :class="$style.nameField"
              required
              @blur="setGuests"
            />
            <v-text-field
              v-if="i !== 1"
              v-model="guests[i - 1].name"
              :label="$t('booking.full_name')"
              placeholder="Ola Nordmann"
              :rules="nameRules"
              :class="$style.nameField"
              required
              @blur="setGuests"
            />
            <v-text-field
              v-model="guests[i - 1].number"
              :label="$t('booking.number')"
              placeholder="95898788"
              number
              :rules="numberRules"
              :class="$style.numberField"
              required
              @blur="setGuests"
            />
            <div :class="$style.emailFieldWrapper">
              <v-text-field
                v-if="i === 1"
                v-model="guests[i - 1].email"
                :label="$vuetify.breakpoint.xs ? $t('booking.emailGeneric') : $t('booking.emailMain')"
                placeholder="ola.n@hotmail.com"
                :rules="emailStrictRules"
                required
              />
              <v-text-field
                v-if="i !== 1"
                v-model="guests[i - 1].email"
                :label="$vuetify.breakpoint.xs ? $t('booking.emailGeneric') : $t('booking.emailOptional')"
                placeholder="ola.n@hotmail.com"
                :rules="emailLooseRules"
              />
            </div>
            <v-checkbox
              v-model="guests[i - 1].isMember"
              class="mx-2"
              :label="$t('booking.member')"
              @change="setGuests"
            ></v-checkbox>
            <v-btn
              :disabled="i === 1"
              :color="$scssVars.globalColorBackgroundLight"
              :class="$style.rmvBtn"
              @click="removeGuestField(i)"
              >x</v-btn
            >
          </div>
        </v-layout>
        <v-btn :class="$style.addBtn" :color="$scssVars.globalColorPrimary" raised="true" @click="addGuestField"
          >+</v-btn
        >
      </v-form>
    </v-layout>
    <h3 :class="$style.price">Total price: {{ price }}kr</h3>
  </v-layout>
</template>

<script lang="ts">
import Vue from 'vue';
import { Guest, BookingSecondStepData } from '../types/booking';
export default Vue.extend({
  data(): BookingSecondStepData {
    return {
      bedsChosen: 1,
      bookingStep: 1,
      guests: [{ name: '', number: '', email: '', isMember: true, isMainBooker: true }],
      numberRules: [
        (e: any) => !!e || 'Number is required',
        (e: any) => e.length >= 8 || 'Number must be 8 characters'
      ],
      nameRules: [(e: any) => !!e || 'Name is required'],
      emailStrictRules: [
        (v: any) => !!v || 'E-mail is required',
        (v: any) => /.+@.+/.test(v) || 'E-mail must be valid'
      ],
      emailLooseRules: [(v: any) => v === '' || /.+@.+/.test(v) || 'E-mail must be valid'],

      edited: false,
      validForm: true,
      memberPrice: 0,
      nonMemberPrice: 0
    };
  },
  computed: {
    step(): number {
      return this.$store.state.booking.step;
    },
    guestField(): Array<Guest> {
      return this.guests;
    },
    beds(): number {
      return this.$store.state.booking.beds;
    },
    price(): number {
      let member = 0;
      let nonMember = 0;
      this.guests.forEach((i) => {
        if (i.isMember) {
          member++;
        } else {
          nonMember++;
        }
      });
      return this.memberPrice * member + this.nonMemberPrice * nonMember;
    }
  },
  watch: {
    validForm: function() {
      this.$store.dispatch('booking/SET_VALID_FORM', this.validForm);
    }
  },

  mounted() {
    this.edited = this.$store.state.booking.edited;
    this.bedsChosen = Number(this.$store.state.booking.beds);
    if (this.edited === true) {
      this.guests = this.$store.state.booking.guests;
    } else {
      this.addGuests();
    }
    this.bookingStep = Number(this.$store.state.booking.step);
    this.$store.dispatch('booking/SET_VALID_FORM', this.validForm);
    this.memberPrice = this.$store.state.koie.koieData.price_member;
    this.nonMemberPrice = this.$store.state.koie.koieData.price_not_member;
  },
  methods: {
    setGuests() {
      this.$store.dispatch('booking/SET_GUESTS', this.guests);
      this.$store.dispatch('booking/SET_EDITED', true);
    },
    addGuests() {
      for (let i = 0; i < this.bedsChosen - 1; i++) {
        this.guests.push({ name: '', number: '', isMember: true, isMainBooker: false });
      }
    },
    addGuestField() {
      if (this.bedsChosen < this.$store.state.booking.availableBeds) {
        this.guests.push({ name: '', number: '', isMember: true, isMainBooker: false });
        this.bedsChosen += 1;
        this.$store.dispatch('booking/SET_BEDS', this.bedsChosen);
      }
    },
    removeGuestField(index: number) {
      if (this.guests.length > 1) {
        this.guests.splice(index - 1, 1);
        this.bedsChosen -= 1;
        this.$store.dispatch('booking/SET_GUESTS', this.guests);
        this.$store.dispatch('booking/SET_BEDS', this.bedsChosen);
      }
    }
  }
});
</script>

<style lang="scss" module>
.separator {
  border-left: solid 4px;
  border-color: #2f3a50;
  display: flex;
  flex-direction: column;
  margin-top: 24px;
}
.container {
  display: flex;
  flex-direction: column;
}

.form,
.guestRegistration {
  padding-left: 16px;
  padding-right: 16px;
  @media only screen and (max-width: 600px) {
    padding-left: 8px;
    padding-right: 8px;
  }
}

.fieldWrapper {
  display: flex;
  width: 100%;
  align-items: center;
}

.emailFieldWrapper {
  width: 50%;
}

.nameField {
  width: 40%;
  padding-right: 10px;
}

.numberField {
  padding-right: 10px;
}
.rmvBtn {
  margin-left: 20px;
  @media only screen and (max-width: 600px) {
    margin-left: 8px;
    padding-left: 0px !important;
    padding-right: 0px !important;
    min-width: 30px !important;
  }
}
.price {
  text-align: center;
  margin-top: 30px;
}

.addBtn {
  margin-top: 16px;
}
</style>
