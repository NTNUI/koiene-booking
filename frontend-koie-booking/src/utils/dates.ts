import dayjs from 'dayjs';
require('dayjs/locale/nb');
dayjs.locale('nb');

export function getDateString(baseDate?: string, addDays?: number): string {
  let resultString: string;
  if (baseDate && addDays) {
    resultString = new Date(new Date(baseDate).getTime() + addDays * 24 * 60 * 60 * 1000).toISOString().substr(0, 10);
  } else if (baseDate) {
    resultString = new Date(baseDate).toISOString().substr(0, 10);
  } else if (addDays) {
    resultString = new Date(new Date().getTime() + addDays * 24 * 60 * 60 * 1000).toISOString().substr(0, 10);
  } else {
    resultString = new Date().toISOString().substr(0, 10);
  }
  return resultString;
}

export function getDate(baseDate?: string, addDays?: number): Date {
  let resultDate: Date;
  if (baseDate && addDays) {
    resultDate = new Date(new Date(baseDate).getTime() + addDays * 24 * 60 * 60 * 1000);
  } else if (baseDate) {
    resultDate = new Date(baseDate);
  } else if (addDays) {
    resultDate = new Date(new Date().getTime() + addDays * 24 * 60 * 60 * 1000);
  } else {
    resultDate = new Date();
  }
  return resultDate;
}

export function addToDate(startDate: string, howMany: number, timeUnit: dayjs.OpUnitType): string {
  const date = dayjs(startDate + 'T00:00:00.000Z');
  return date.add(howMany, timeUnit).format('YYYY-MM-DD');
}

export function formatDate(dateISO: string, formatString: string): string {
  const date = dayjs(dateISO + 'T00:00:00.000Z');
  return date.format(formatString);
}
