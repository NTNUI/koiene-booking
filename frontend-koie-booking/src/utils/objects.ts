export const clone = (object: any) => Object.assign({}, object);

export const renameKey = (object: any, key: string, newKey: string): any => {
  const value = object[key];
  delete object[key];
  object[newKey] = value;
  return object;
};

export const sortObjectByKey = (unordered: any) => {
  let ordered: any;
  ordered = {};
  Object.keys(unordered)
    .sort()
    .forEach(function(key) {
      ordered[key] = unordered[key];
    });
  return ordered;
};
