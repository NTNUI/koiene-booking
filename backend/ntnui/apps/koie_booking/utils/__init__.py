from koie_booking.utils.mail_utils import send_confirmation_mail, send_koie_information_mail
from koie_booking.utils.date_utils import get_daterange
