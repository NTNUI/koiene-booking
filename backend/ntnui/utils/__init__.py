from ntnui.utils.pagination import PageSize7, PageSize20, PageSize100
