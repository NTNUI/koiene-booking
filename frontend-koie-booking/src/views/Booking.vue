<template>
  <v-container class="global-max-width">
    <v-row>
      <v-col cols="12" sm="8" class="mx-auto" :class="$style.mainColumn">
        <v-stepper v-model="step" alt-labels :class="$style.stepper">
          <v-stepper-header :class="$style.stepperHeader">
            <v-stepper-step :complete="step > 1" :step="1">
              {{ labels[0] }}
            </v-stepper-step>
            <v-divider></v-divider>
            <v-stepper-step :complete="step > 2" :step="2">
              {{ labels[1] }}
            </v-stepper-step>
            <v-divider></v-divider
            ><v-stepper-step :complete="step > 3" :step="3">
              {{ labels[2] }}
            </v-stepper-step>
            <v-divider></v-divider
            ><v-stepper-step :step="4">
              {{ labels[3] }}
            </v-stepper-step>
          </v-stepper-header>
          <v-layout>
            <LoadingSpinner v-if="isLoading" />
            <BookingFirstStep v-else-if="step === 1" />
            <BookingSecondStep v-else-if="step === 2" />
            <BookingThirdStep v-else-if="step === 3" />
            <BookingFourthStep v-else-if="step === 4" />
          </v-layout>
        </v-stepper>
        <div v-show="!apiError" :class="$style.btnWrapper">
          <v-btn
            v-if="step > 1 && step < 4"
            :color="this.$scssVars.globalColorPrimary"
            raised="true"
            :dark="true"
            data-test="btnPrev"
            @click="prevStep"
            >{{ $t('booking.back_button') }}</v-btn
          >
          <v-btn
            v-if="step < 3"
            :disabled="!checkValid || disableBooking"
            :color="this.$scssVars.globalColorPrimary"
            :class="$style.nextStepButton"
            raised="true"
            :dark="true"
            data-test="btnNext"
            @click="nextStep"
            >{{ $t('booking.next_button') }}</v-btn
          >
          <v-btn
            v-if="step === 4"
            :color="this.$scssVars.globalColorPrimary"
            :class="$style.nextStepButton"
            raised="true"
            :dark="true"
            data-test="btnNext"
            @click="done"
            >{{ $t('booking.done_button') }}</v-btn
          >
          <v-alert
            v-if="disableBooking"
            v-model="disableBooking"
            transition="expand-transition"
            :class="$style.errorMsg"
            type="info"
            @click="error = false"
          >
            {{ $t('avalanche.error_msg') }}
          </v-alert>
        </div>
      </v-col>
    </v-row>
  </v-container>
</template>

<script lang="ts">
import BookingFirstStep from '../components/BookingFirstStep.vue';
import BookingSecondStep from '../components/BookingSecondStep.vue';
import BookingThirdStep from '../components/BookingThirdStep.vue';
import BookingFourthStep from '../components/BookingFourthStep.vue';
import LoadingSpinner from '../components/LoadingSpinner.vue';
import { BookingData } from '../types/booking';
import Vue from 'vue';
export default Vue.extend({
  name: 'Booking',
  components: {
    BookingFirstStep,
    BookingSecondStep,
    BookingThirdStep,
    BookingFourthStep,
    LoadingSpinner
  },

  data(): BookingData {
    return {
      step: 1,
      labels: [
        this.$i18n.t('booking.step1'),
        this.$i18n.t('booking.step2'),
        this.$i18n.t('booking.step3'),
        this.$i18n.t('booking.step4')
      ]
    };
  },
  computed: {
    dateFrom(): string {
      return this.$store.state.booking.dateFrom;
    },
    dateTo(): string {
      return this.$store.state.booking.dateTo;
    },
    error(): boolean {
      return this.$store.state.koie.error;
    },
    createBookingError(): boolean {
      return this.$store.state.booking.error;
    },
    isLoading(): boolean {
      return this.$store.state.booking.isLoading;
    },
    checkValid(): string {
      return this.$store.state.booking.validForm;
    },
    koieTitle(): string {
      return this.$store.state.koie.koieData.name || this.$route.params.id;
    },
    disableBooking(): boolean {
      return this.$store.state.avalanche.disableBooking;
    },
    apiError(): boolean {
      return this.$store.state.koie.error;
    },
    bookingStep(): boolean {
      return this.$store.state.booking.step;
    }
  },
  watch: {
    dateFrom: function() {
      this.fetchWarningData();
    },
    dateTo: function() {
      this.fetchWarningData();
    },
    apiError(): boolean {
      return this.$store.state.koie.error;
    },
    bookingStep: function() {
      this.step = this.$store.state.booking.step;
      if (this.$store.state.booking.step === 1) {
        this.resetBookingInfo();
      }
    }
  },
  mounted() {
    //resetting everything when component mounts
    this.$store.dispatch('booking/SET_STEP', 1);
    this.step = 1;
    this.$store.dispatch('booking/SET_VALID_FORM', true);
    this.resetBookingInfo();
    this.$store.dispatch('booking/SET_EDITED', false);
  },
  methods: {
    async fetchWarningData() {
      if (this.$store.state.avalanche.koierToCheck.includes(this.koieTitle)) {
        const values = {
          latitude: this.$store.state.koie.koieData.latitude,
          longitude: this.$store.state.koie.koieData.longitude,
          language: 2,
          dateFrom: this.$store.state.booking.dateFrom,
          dateTo: this.$store.state.booking.dateTo
        };
        await this.$store.dispatch('avalanche/FETCH_AVALANCHE_LEVELS', values);
        const warningData = await this.$store.state.avalanche.warningData;
        for (let e in warningData) {
          if (warningData[e].warningLevel > 2) {
            this.$store.dispatch('avalanche/DISABLE_BOOKING', true);
            return;
          } else {
            this.$store.dispatch('avalanche/DISABLE_BOOKING', false);
          }
        }
      }
    },
    async resetBookingInfo() {
      const guests = [{ name: '', number: '', email: '', isMember: true }];
      for (let i = 0; i < Number(this.$store.state.booking.beds) - 1; i++) {
        guests.push({ name: '', number: '', email: '', isMember: true });
        this.$store.dispatch('booking/SET_GUESTS', guests);
      }

      if (this.$store.state.koie.koieData.name.length === 0) {
        await this.$store.dispatch('koie/FETCH_DATA', this.koieTitle);
      }
      this.fetchWarningData();
      this.$store.dispatch('booking/SET_GUESTS', [{ name: '', number: '', email: '', isMember: true }]);
    },
    async nextStep() {
      if (this.step === 2) {
        const values = {
          koie: await this.$store.state.koie.koieData.name.toLowerCase(),
          arrival_date: await this.$store.state.booking.dateFrom,
          departure_date: await this.$store.state.booking.dateTo,
          guests_member: await this.$store.state.booking.numberOfMembers,
          guests_not_member: await this.$store.state.booking.numberOfNonMembers
        };
        this.$store.dispatch('booking/CREATE_BOOKING', values);
      }
      if (this.step === 2 && this.createBookingError) {
        this.step = 1;
        this.$store.dispatch('booking/SET_STEP', Number(this.$store.state.booking.step) + 1);
      } else {
        this.step < 4 && this.$store.dispatch('booking/SET_STEP', Number(this.$store.state.booking.step) + 1);
        this.step = Number(this.$store.state.booking.step);
      }
    },
    prevStep() {
      this.step >= 1 && this.$store.dispatch('booking/SET_STEP', Number(this.$store.state.booking.step) - 1);
      this.step = Number(this.$store.state.booking.step);
      if (this.step === 1) {
        this.$store.dispatch('booking/SET_VALID_FORM', true);
      }
    },
    firstStep() {
      this.$store.dispatch('booking/SET_STEP', 1);
      this.step = 1;
    },
    done() {
      this.$router.push(`/`);
    }
  }
});
</script>

<style lang="scss" module>
.mainColumn {
  @media only screen and (max-width: 600px) {
    padding-left: 0px;
    padding-right: 0px;
  }
}

.stepper {
  min-height: 80vh;
  box-shadow: none;
  background: none !important;
  @media only screen and (min-width: 1500px) {
    min-height: 60vh;
  }
}
.stepperHeader {
  box-shadow: none;
}

.btnWrapper {
  display: flex;
  align-items: center;
  justify-content: center;
  bottom: 5%;
}
.nextStepButton {
  margin-left: 20px;
}
.errorMsg {
  margin: 10px 16px 10px 16px;
}
</style>
