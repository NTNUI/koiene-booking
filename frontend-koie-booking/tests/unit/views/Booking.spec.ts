import Vue from 'vue';
import Vuetify from 'vuetify';
import flushPromises from 'flush-promises';
import mockAxios from 'jest-mock-axios';

Vue.use(Vuetify);

// Components or views
import Booking from '@/views/Booking.vue';
import BookingFirstStep from '@/components/booking/BookingFirstStep.vue';
import BookingSecondStep from '@/components/booking/BookingSecondStep.vue';
import { createWrapper } from '../utils';

describe('View Booking.vue', () => {
  it('Matches snapshot', async () => {
    const wrapper = createWrapper(Booking, {
      propsData: {
        allKoier: false,
        row: false,
      },
      mocks: {
        $route: {
          params: { id: 'flåkoia' },
        },
      },
    });

    const responseObj = { data: { koie: { name: 'flåkoia' } } };
    mockAxios.mockResponse(responseObj);
    await flushPromises();

    expect(wrapper).toMatchSnapshot();
  });

  it('Button_next renders next bookingStep', async () => {
    const wrapper = createWrapper(Booking, {
      propsData: {
        allKoier: false,
        row: false,
      },
      mocks: {
        $route: {
          params: { id: 'flåkoia' },
        },
      },
    });

    const responseObj = { data: { koie: { name: 'flåkoia' } } };
    mockAxios.mockResponse(responseObj);
    await flushPromises();

    wrapper.find('[data-test="btnNext"]').trigger('click');

    expect(wrapper.contains(BookingSecondStep)).toBe(true);
    expect(wrapper.contains(BookingFirstStep)).toBe(false);
  });

  it('Button_prev renders prev bookingStep', async () => {
    const wrapper = createWrapper(Booking, {
      propsData: {
        allKoier: false,
        row: false,
      },
      mocks: {
        $route: {
          params: { id: 'flåkoia' },
        },
      },
    });

    const responseObj = { data: { koie: { name: 'flåkoia' } } };
    mockAxios.mockResponse(responseObj);
    await flushPromises();

    wrapper.find('[data-test="btnNext"]').trigger('click');
    wrapper.find('[data-test="btnPrev"]').trigger('click');
    await flushPromises();

    expect(wrapper.contains(BookingFirstStep)).toBe(true);
    expect(wrapper.contains(BookingSecondStep)).toBe(false);
  });
});
