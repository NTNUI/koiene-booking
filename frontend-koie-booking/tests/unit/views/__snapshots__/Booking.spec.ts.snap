// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Booking.vue Matches snapshot 1`] = `
<div class="container global-max-width">
  <div class="row">
    <div class="mx-auto col-sm-8 col-12 mainColumn">
      <div class="v-stepper v-stepper--alt-labels theme--light stepper">
        <div class="v-stepper__header stepperHeader">
          <div class="v-stepper__step v-stepper__step--active"><span class="v-stepper__step__step primary">1</span>
            <div class="v-stepper__label">
              Koie
            </div>
          </div>
          <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
          <div class="v-stepper__step v-stepper__step--inactive"><span class="v-stepper__step__step">2</span>
            <div class="v-stepper__label">
              Register
            </div>
          </div>
          <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
          <div class="v-stepper__step v-stepper__step--inactive"><span class="v-stepper__step__step">3</span>
            <div class="v-stepper__label">
              Pay
            </div>
          </div>
          <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
          <div class="v-stepper__step v-stepper__step--inactive"><span class="v-stepper__step__step">4</span>
            <div class="v-stepper__label">
              Receipt
            </div>
          </div>
        </div>
        <div class="layout">
          <div class="layout dark container">
            <h1 class="heading">Reservation for flåkoia</h1>
            <div class="layout separator">
              <h3>Dates</h3>
              <div class="layout px-1">
                <div class="container row row">
                  <div class="v-input v-input--is-label-active v-input--is-dirty theme--dark v-text-field rowTextFieldArrival">
                    <div class="v-input__prepend-outer">
                      <div class="v-input__icon v-input__icon--prepend"><i aria-hidden="true" class="v-icon notranslate material-icons theme--dark">event</i></div>
                    </div>
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <div class="v-text-field__slot"><label for="input-21" class="v-label v-label--active theme--dark" style="left: 0px; position: absolute;">Arrival</label><input id="input-21" type="text"></div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--dark"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
                      </div>
                    </div>
                  </div>
                  <div class="v-menu">
                    <!---->
                  </div>
                  <div class="v-menu">
                    <!---->
                  </div>
                  <div class="v-input v-input--is-label-active v-input--is-dirty theme--dark v-text-field rowTextFieldDeparture bookingTabletRemovePadding">
                    <div class="v-input__prepend-outer">
                      <div class="v-input__icon v-input__icon--prepend"><i aria-hidden="true" class="v-icon notranslate material-icons theme--dark">event</i></div>
                    </div>
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <div class="v-text-field__slot"><label for="input-26" class="v-label v-label--active theme--dark" style="left: 0px; position: absolute;">Departure</label><input id="input-26" type="text"></div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--dark"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
                      </div>
                    </div>
                  </div>
                  <div class="divBeds">
                    <!---->
                    <!---->
                    <!---->
                  </div>
                  <!---->
                </div>
              </div>
            </div>
            <div class="layout separator">
              <h3>Bed Reservation</h3>
              <div class="layout availableBeds"><i aria-hidden="true" class="v-icon notranslate mdi mdi-bed theme--light icon"></i>
                <p class="availableBedsText">
                  Infinity / available beds
                </p>
              </div>
              <div class="layout chooseBeds">
                <p class="chooseText" style="margin-bottom: 0px;">Choose number of beds: </p>
                <div class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outlined bedPicker">
                  <div class="v-input__control">
                    <div class="v-input__slot">
                      <fieldset aria-hidden="true">
                        <legend><span>​</span></legend>
                      </fieldset>
                      <div class="v-text-field__slot"><input min="1" max="Infinity" id="input-31" type="number"></div>
                    </div>
                  </div>
                </div>
              </div>
              <!---->
            </div>
          </div>
        </div>
      </div>
      <div class="btnWrapper">
        <!----> <button type="button" class="v-btn v-btn--contained theme--dark v-size--default nextStepButton" raised="true" data-test="btnNext"><span class="v-btn__content">next</span></button>
        <!---->
        <!---->
      </div>
    </div>
  </div>
</div>
`;
